### ISSUE TITLE
[Bug]: Links with long URLs are hard to edit

### ISSUE THREAD

**Issue by scruffian** [Automattic/wp-calypso#69563]
### Quick summary

When links have long URLs its not obvious where the edit button is:

<img width="727" alt="Screenshot 2022-10-28 at 12 06 57" src="https://user-images.githubusercontent.com/275961/198573161-6e80d67e-25ff-4f93-931a-cbfe2a55c2a0.png">

if you scroll inside that box to the right then it is there:

<img width="763" alt="Screenshot 2022-10-28 at 12 07 04" src="https://user-images.githubusercontent.com/275961/198573224-cdaa7fc6-3151-4c09-8695-5b12e830aa56.png">

Note that this is different on WordPress.org
<img width="469" alt="Screenshot 2022-10-28 at 12 07 10" src="https://user-images.githubusercontent.com/275961/198573259-3835dfbf-80ea-43c4-ba2b-0c889cd09188.png">

### Steps to reproduce

1. Add a link with a long URL
2. Go to edit the link
3. Notice that the edit button is hidden and you need to scroll to find it

### What you expected to happen

I expected it to work the same as in a dotorg environment.

### What actually happened

The edit button was hard to find.

### Context

Building a site

### Platform (Simple, Atomic, or both?)

Simple

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

No but the platform is still usable

### Workaround details

Once you know to scroll it's ok.

**Comment by TimBroddin** [comment]
I've taken a look at this problem and I notice two things:

1) On WPCOM the title isn't retrieved and defaults to the full URL, on .org the title is "Before you continue" (Google's privacy wall)
2) The title field of the edit UI doesn't break.

### Title not being retrieved
As for the title that's not being retrieved. The block calls `https://public-api.wordpress.com/wp-block-editor/v1/sites/<siteid>/url-details` on WPCOM and receives:

```json
{
  "body": {
    "title": "",
    "icon": "https://ssl.gstatic.com/travel-trips-fe/travel_logo_32.png",
    "description": "",
    "image": ""
  },
  "status": 200,
  "headers": {
    "Allow": "GET"
  }
}
```

On .org it calls `index.php?rest_route=wp-block-editor/v1/url-details?url=` and receives:

```json
{"title":"Before you continue","icon":"https://www.google.com/favicon.ico","description":"","image":""}
````

### The title field doesn't break
The title field doesn't break on long words, and thus overflows. 

<img width="780" alt="CleanShot 2022-10-28 at 14 30 03@2x" src="https://user-images.githubusercontent.com/528287/198587639-4a17aa2a-ecac-44dd-844c-487d355caec7.png">

Fixing this could be as easy as adding this CSS:

```css
.block-editor-link-control__search-item .block-editor-link-control__search-item-title {
  word-break: break-word;
}
```

<img width="800" alt="CleanShot 2022-10-28 at 14 31 58@2x" src="https://user-images.githubusercontent.com/528287/198587571-2fe38408-2098-497b-b948-eff3887d8134.png">

This issue is present on both .com/.org but does not show on .org in this case because it is able to retrieve the title.

**Comment by scruffian** [comment]
It seems like the best fix would be to ensue that the title is returned then.

**Comment by TimBroddin** [comment]
I submitted [a PR](https://github.com/WordPress/gutenberg/pull/45387) to Jetpack to wrap the title, because this bug would also occur if the site would return a long title with no spaces.

I created [a ticket](https://github.com/Automattic/wp-calypso/issues/69575) for our team to look into the title fetching problem. I did some initial testing, and I assume that for some reason Google is blocking requests from us. But, that's an assumption and I will further investigate next week.

Another solution would be to fall back to "(no title)" when there is no title returned. URLs will often be too long for the quite narrow field.

**Comment by TimBroddin** [comment]
I did some further research into the no title returned at `google.com/travel` problem.  We currently cap the max response size at 150 KB. The HTML Google returns us, contains many inline styles, making the `<title>` tag fall outside this 150 KB. 

We could fix this by raising this max response size, but this will also increase network usage. This bug only happens in very select cases (HTML responses where the `<title>` tags is not in the first 150KB), so I don't think this is worth is. Especially since the [UI fix](https://github.com/WordPress/gutenberg/pull/45387) makes the fallback usable. 

_WordPress.org has the same max response size, but gets served different HTML. I tried replicating this behavior by playing around with different User-Agents, but couldn't reproduce._
